src/compiler/ast/module/SourceFile.ts

```diff
@@ -744,6 +744,40 @@ export class SourceFile extends SourceFileBase<ts.SourceFile> {
         return this;
     }
 
+    /**
+     * Removes all unused declarations like interfaces, classes, enums, functions, variables, parameters,
+     * methods, properties, imports, etc. from this file.
+     *
+     * WARNING! This will forget all the nodes in the file! It's best to do this after you're all done with the file.
+     */
+    removeUnusedDeclarations(formatSettings: FormatCodeSettings = {}, userPreferences: UserPreferences = {}) {
+        const sourceFile = this;
+        // heads up: calling it twice removes more unused variables, like things referenced in unused things or named variables
```

Comment:
Doesn't that mean it might need to be called 3 times or 4 or ...etc...? :)

Reply:
Yes, I think so :) Perhaps is better to just call it once? I think is not incorrect not to remove a variable that get's unused after others are removed. I could research a bit more about it.

Reply:
Yes, it should only be called once.